contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -927,6 +928,177 @@ checksum(PG_FUNCTION_ARGS)
         PG_RETURN_INT32(result);
 }
 
+Datum
+object_id(PG_FUNCTION_ARGS)
+{	
+	char 			*db_name = "";
+	char 			*schema_name = "";
+	char 			*physical_schema_name;
+	char 			*object_name;
+	bool 			is_temp_object;
+	Oid 			schema_oid;
+	Oid 			result = InvalidOid;
+	int 			i;
+	char 			*input = lowerstr(text_to_cstring(PG_GETARG_TEXT_P(0)));
+	char 			*object_type = text_to_cstring(PG_GETARG_TEXT_P(1));
+	List 			*splited_object_name;
+
+	/* strip trailing whitespace */
+	i = strlen(input);
+	while (i > 0 && isspace((unsigned char) input[i - 1]))
+		input[--i] = '\0';
+	
+	/* resolve the three part name */
+	splited_object_name = split_object_name(input);
```

Comment:
this looks like a functionality needed else where as well. Could you please search the code if we are splitting three part name at other places as well and reuse that code

Reply:
Earlier there was function written in sql which was doing the same, I have not found any similar function in our code which can be used.